python/tester/upload_from_repo.py

```diff
@@ -331,22 +332,8 @@ def _upload_one_image(temp_dir, group_id, butler, ref):
                     )
                 sidecar_uploaded = True
 
-        # Each ref is done separately because butler.retrieveArtifacts does not preserve the order.
-        transferred = butler.retrieveArtifacts(
-            [ref],
-            transfer="copy",
-            preserve_path=False,
-            destination=temp_dir,
-        )
-        if len(transferred) != 1:
-            _log.error(
-                f"{ref} has multiple artifacts and cannot be handled by current implementation"
-            )
-            for transfer in transferred:
-                os.remove(transfer.path)
-            return
-
-        path = transferred[0].path
+        path = os.path.join(temp_dir, uri.basename())
+        ResourcePath(path).transfer_from(uri, transfer="copy")
```

Comment:
I'm not entirely sure what this code is trying to do, but it looks like it's downloading a file from remote to posix path, optionally extracting a sidecar json, and then uploading the raw it to a new location. Is that correct?

Why isn't this doing a direct transfer from one location to another using ResourcePath? Secondly, why isn't the JSON header extracted in the remote using astropy.io.fits with ResourcePath.to_fsspec? Would simplify the code a lot.

Reply:
Why are we modifying the headers of raw files?

Reply:
Because we are giving them new group ID, new exposure ID, etc to fake new exposures.  The script sends a next_visit event with that metadata, and the prompt service will attempt to process the matching exposure (via the headers).